### DISCUSSION TITLE
Example for reference child properties (not collections)

### DISCUSSION THREAD

**Discussion by vijaygill** [MarimerLLC/csla#2333] [Questions] [answered]
Hi,

Where can I find an example to show following relationship for example - Person class having a property for State and Gender classes.
Also the state / gender properties can be changed via UI (say Blazor).
I tried looking in the examples and I could not find any, may be I missed it or did not search proiperly.

Thanks

**Comment by rockfordlhotka** [comment, marked as answer]
I suspect you are looking for the `RoleList` and `Role` property concepts in the Samples/ProjectTracker example?

Something where you have a read-only, cached, collection of valid values, and then that's used by a rule as people are assigned to projects?

**Reply by vijaygill** [reply]
Thanks @rockfordlhotka !!! That'swhat I was looking for. I will read-up more on this.

Actually I want to use ReadOnlyBase + ReadOnlyListBase so that I can use richer UI. also you book seems to suggest the usage of the same as NVL seems to be for legacy purposes only.

BTW - I have both your books (2005 and 2008), all you CSLA 4 ebooks and also the Blazor book (all purchased from your online store). I have been reading those as time permits and need arises.
